clang/test/Driver/range.c

```diff
@@ -0,0 +1,29 @@
+// Test range options for complex multiplication and division.
+
+// RUN: %clang -### -target x86_64 -fcx-limited-range -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=LMTD %s
+
+// RUN: %clang -### -target x86_64 -fcx-fortran-rules -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=FRTRN %s
+
+// RUN: %clang -### -target x86_64 -fcx-limited-range \
+// RUN: -fcx-fortran-rules -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=WARN1 %s
+
+// RUN: %clang -### -target x86_64 -fcx-fortran-rules \
+// RUN: -fcx-limited-range  -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=WARN2 %s
+
+// RUN: %clang -### -target x86_64 -ffast-math -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=FRTRN %s
+
+// RUN: %clang -### -target x86_64 -ffast-math -fcx-limited-range -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=LMTD %s
+
+// RUN: %clang -### -target x86_64 -fcx-limited-range -ffast-math -c %s 2>&1 \
+// RUN:   | FileCheck --check-prefix=FRTRN %s
+
+// LMTD: -complex-range=cx_limited
+// FRTRN: -complex-range=cx_fortran
```

Comment:
I think you also want LMTD-NOT and FRTRN-NOT lines to make sure that you aren't specifying multiple `-complex-range=*` arguments at the same time.